x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/function/Function.java

```diff
@@ -42,6 +43,11 @@ public Nullability nullable() {
         return Expressions.nullable(children());
     }
 
+    /** Return true if this function can be executed under the provided {@link XPackLicenseState}, otherwise false.*/
+    public boolean checkLicense(XPackLicenseState state) {
```

Comment:
At first I was a bit concerned about passing the entire XPackLicenseState down here, but your tests below make it look reasonable.